Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -455,14 +487,14 @@ protected override void ProcessRecord()
                     else
                     {
                         TablesResource.ListRequest checkTables =
-                            new TablesResource.ListRequest(Service, Project, InputObject.DatasetReference.DatasetId);
+                            new TablesResource.ListRequest(Service, Project, Dataset);
                         var tableResponse = checkTables.Execute();
                         if (tableResponse.TotalItems == 0)
                         {
                             response = request.Execute();
                         }
                         else if (ShouldContinue(
-                            GetConfirmationMessage(InputObject, tableResponse.TotalItems),
+                            GetConfirmationMessage(Dataset, tableResponse.TotalItems),
```

Comment:
I do.  It's checked above the surrounding else (line 482).  That way if -Force is supplied, we don't waste time doing a check to see if the table is empty first.  It's just deleted anyway.